toolkit/offchain/src/plutus_script.rs

```diff
@@ -0,0 +1,68 @@
+use anyhow::{anyhow, Context};
+use cardano_serialization_lib::{Address, LanguageKind, NetworkIdKind, PlutusData};
+use plutus::ToDatum;
+use sidechain_domain::{MainchainAddressHash, PolicyId};
+use uplc::ast::{DeBruijn, Program};
+
+use crate::{csl::*, untyped_plutus::*};
+
+/// Wraps a Plutus script cbor
+pub struct PlutusScript {
+	pub bytes: Vec<u8>,
+	pub language: LanguageKind,
+}
+
+impl PlutusScript {
+	pub fn from_cbor(cbor: &[u8], language: LanguageKind) -> Self {
+		Self { bytes: cbor.into(), language }
+	}
+	pub fn from_wrapped_cbor(cbor: &[u8], language: LanguageKind) -> anyhow::Result<Self> {
+		Ok(Self::from_cbor(&unwrap_one_layer_of_cbor(cbor)?, language))
+	}
+
+	pub fn apply_data(self, data: impl ToDatum) -> Result<Self, anyhow::Error> {
```

Comment:
Are versions in untyped_plutus.rs still required?

Reply:
They're not. I removed them and only left casts and unwrapping there.